include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -251,7 +260,13 @@ Operator<dim, Number>::fill_matrix_free_data(MatrixFreeData<dim, Number> & matri
   }
 
   // dealii::Quadrature
-  matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.degree + 1), get_quad_name());
+  if(this->grid->triangulation->all_reference_cells_are_hyper_cube())
+    matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.degree + 1), get_quad_name());
+  else if(this->grid->triangulation->all_reference_cells_are_simplex())
+    matrix_free_data.insert_quadrature(dealii::QGaussSimplex<dim>(param.degree + 1),
+                                       get_quad_name());
+  else
+    AssertThrow(false, ExcNotImplemented());
```

Comment:
I want to ask a question here. Why are we using `QGauss<1>`  and not `QGauss<dim>` for the hypercube case? I had to introduce the same difference for the Poisson case.

This is also the reason why we can't use the following function: https://www.dealii.org/developer/doxygen/deal.II/classReferenceCell.html#a56bb8396fda394d712fc4f19de4b02c4

Reply:
> I want to ask a question here. Why are we using `QGauss<1>` and not `QGauss<dim>` for the hypercube case? I had to introduce the same difference for the Poisson case.
> 
> This is also the reason why we can't use the following function: https://www.dealii.org/developer/doxygen/deal.II/classReferenceCell.html#a56bb8396fda394d712fc4f19de4b02c4

You can simply switch to `QGauss<dim>`. `MatrixFree` will detect the tensor-product structure internally and will create the 1D quadrature for you.

Reply:
should I change it to `dim` here?

Reply:
> should I change it to dim here?

I don't have an opinion on that. I guess the `MatrixFreeWrapper` will do the conversion anyway interanlly?